### ISSUE TITLE
A3 interoperability

### ISSUE THREAD

**Issue by mf59816** [liqd/thentos#110]
This is a big ticket, and will probably require more explanation from my part, but it's a start.

Each of the following steps should be self-contained.  Changing adhocracy code will not be necessary until we reach step 3.

1. run thentos as a trivial rest proxy for adhocracy (trivial == "does not change behavior, only listens in.")
2. catch user registration requests, and feed thentos db with users.
3. make adhocracy (1) accept session tokens without checking, and (2) fill its user resources via rest api calls to thentos.
4. sub-services: different instances on the same adhocracy installation should be different services for thentos, and get different pseudonyms for the same user.

the last point certainly requires more thinking and should probably be considered an outlook rather than part of this ticket.

we may also want to start opening sub-tickets on this.

**Comment by mf59816** [comment]
opened https://github.com/liqd/adhocracy3.mercator/issues/1349

**Comment by fisx** [comment]
ChristianS: giving you the lead on this.

(Item 4. from the list above should become a new ticket once this is done.  We may want to do SSO first, too.)

**Comment by ChristianSi** [comment]
I have Thentos running as Proxy between A3 front- and backend, but there are still various problems to solve until anything useful will be possible. E.g.

1. The proxy refuses requests if the user is not logged in ("ActionErrorThentos NoSuchThentosSession"). But of course, such requests are necessary for user creation and user login as well as for anonymous browsing!

2. Our proxy doesn't understand the A3 batch API (batch request). Sooner or later we may have to tackle them, and that's not a small task.

I'll work on (1) first.

**Comment by ChristianSi** [comment]
The happy flow is working now. I don't think we'll need batch requests (at least for the time being), since user creation, account activation and user login are never sent in batch requests by the A3 frontend.

I'm looking at the error cases now -- it may be possible to mimic the typical A3 backend error messages so that the A3 frontend doesn't need to be changed.

**Comment by ChristianSi** [comment]
I believe that with the completion of #168 this task is complete. Except for (4), sub-services or personas, which I would see as a different task, especially as the requirements for that one are still very unclear.

@fisx If you agree, please close this ticket.

**Comment by fisx** [comment]
@ChristianSi: agreed (:
